File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.js
/**
 * @param {number[]} cardPoints
 * @param {number} k
 * @return {number}
 */
var maxScore = function(cardPoints, k) {
  const n= cardPoints.length;
  let total =0;
  for(i=0; i<k; i++){
    total+=cardPoints[i];
  }  
  let maxPoints= total;
  //Shifting
  for (let i=0; i<k; i++){
    total-=cardPoints[k-1-i];
    total+=cardPoints[n-1-i];
    maxPoints = Math.max(maxPoints, total);
  }
  return maxPoints;
};

File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.java
class Solution {
    public int maxScore(int[] cardPoints, int k) {
        //get the total number of cards
        int n = cardPoints.length;
        //calculate sum of first k cards from the front
        int total =0;
        for(int i =0; i<k; i++){
            total+=cardPoints[i];
        }
        //Store the max score 
        int maxPoints= total;
        //Slide the window remove from front and add it in the back
        for(int i =0; i<k; i++){
            //Subtract Card from the front
            total -=cardPoints[k-1-i];
            //Add Caerd from back 
            total +=cardPoints[n-1-i];
            maxPoints = Math.max(maxPoints, total);
        }
        //return best possible score
        return maxPoints;
    }
}